### ISSUE TITLE
ttnn.repeat works incorrectly when input tensor has UINT32 datatype

### ISSUE THREAD

**Issue by amalbasaTT** [tenstorrent/tt-metal#19776]
**Describe the bug**
ttnn.repeat works incorrectly when input tensor has ttnn.uint32 datatype. 

Maximum relative error is 8388608.0 when input has spec:
- inputs are all ones
- input shape: [2, 1, 1, 7]
- datatype: ttnn.uint32
- layout: ttnn.TILE_LAYOUT
ttnn.repeat produces tensor with all elements 8388609
This spec is found in CLIP model and is one of the causes of low accuracy in that model.

**To Reproduce**
Run unit test:
```
from loguru import logger
import random
import pytest
import torch
import ttnn
import traceback
from models.utility_functions import torch_random
from tests.ttnn.utils_for_testing import assert_with_pcc
aten = torch.ops.aten


def max_relative_error(expected, calculated):
    assert expected.shape == calculated.shape, "Tensors must have the same shape"
    if expected.dtype == torch.bool:
        expected = expected.to(torch.int)
    if calculated.dtype == torch.bool:
        calculated = calculated.to(torch.int)
    relative_error = torch.abs(expected - calculated) / (torch.abs(expected) + 1e-8)
    mre = torch.max(relative_error).item()
    inds = torch.where(relative_error == mre)
    return mre, inds


def run_repeat_tests(
    input_shape, 
    dtype, 
    dlayout,
    repetition_vector,
    device,
):
    # All ones is input found in CLIP model
    # x = torch.ones(size=input_shape[0], dtype=torch.int32)
    x = torch.randint(-100, 100, input_shape[0], dtype=torch.bfloat16)

    try:
        ref_value = aten.repeat.default(x, repetition_vector[0])
        
        tt_x = ttnn.from_torch(x, dtype=dtype[0], layout=dlayout[0], device=device)

        tt_result = ttnn.repeat(tt_x, repetition_vector[0])
        tt_result = ttnn.to_torch(tt_result)
    except Exception as e:
        logger.warning(f"Test execution crashed: {e}")
        print(traceback.format_exc())
        raise e

    assert len(tt_result.shape) == len(ref_value.shape)
    assert tt_result.shape == ref_value.shape
    mre, inds = max_relative_error(ref_value, tt_result)
    assert mre < (0.0 + 1e-8), f"Maximum relative error is {mre}, {tt_result[inds]}"


test_sweep_args = [
    (
        [[2, 1, 1, 7]],
        [ttnn.uint32],
        [ttnn.TILE_LAYOUT],
        [[1, 1, 7, 1]],
    ),
]


@pytest.mark.parametrize(
    "input_shape, dtype, dlayout, repetition_vector",
    (test_sweep_args),
)
def test_repeat(input_shape, dtype, dlayout, repetition_vector, device):
    run_repeat_tests(input_shape, dtype, dlayout, repetition_vector, device)

```
